src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp

```diff
@@ -1055,82 +1055,94 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_WriteThreadNetw
+            if (otRole == OT_DEVICE_ROLE_DISABLED)
+            {
+                routingRole = RoutingRoleEnum::kUnspecified;
+            }
+            else if (otRole == OT_DEVICE_ROLE_DETACHED)
+            {
+                routingRole = RoutingRoleEnum::kUnassigned;
+            }
+            else if (otRole == OT_DEVICE_ROLE_ROUTER)
             {
-                routingRole = RoutingRoleEnum::kEndDevice;
+                routingRole = RoutingRoleEnum::kRouter;
+            }
+            else if (otRole == OT_DEVICE_ROLE_LEADER)
+            {
+                routingRole = RoutingRoleEnum::kLeader;
+            }
+            else if (otRole == OT_DEVICE_ROLE_CHILD)
+            {
+                otLinkModeConfig linkMode = otThreadGetLinkMode(mOTInst);
+
+                if (linkMode.mRxOnWhenIdle)
+                {
+                    routingRole = RoutingRoleEnum::kEndDevice;
 #if CHIP_DEVICE_CONFIG_THREAD_FTD
-                if (otThreadIsRouterEligible(mOTInst))
+                    if (otThreadIsRouterEligible(mOTInst))
+                    {
+                        routingRole = RoutingRoleEnum::kReed;
+                    }
+#endif
+                }
+                else
                 {
-                    routingRole = RoutingRoleEnum::kReed;
+                    routingRole = RoutingRoleEnum::kSleepyEndDevice;
                 }
-#endif
-            }
-            else
-            {
-                routingRole = RoutingRoleEnum::kSleepyEndDevice;
             }
-        }
 
-        err = encoder.Encode(routingRole);
+            return pAttrEncoder->Encode(routingRole);
+        });
     }
     break;
 
     case ThreadNetworkDiagnostics::Attributes::NetworkName::Id: {
-        const char * networkName = otThreadGetNetworkName(mOTInst);
-        err                      = encoder.Encode(CharSpan::fromCharString(networkName));
+        err = encoder.EncodeNullableOnCondition(mIsAttached, [this](const auto pAttrEncoder) -> CHIP_ERROR {
+            const char * networkName = otThreadGetNetworkName(mOTInst);
+            return pAttrEncoder->Encode(CharSpan::fromCharString(networkName));
+        });
     }
     break;
 
     case ThreadNetworkDiagnostics::Attributes::PanId::Id: {
-        uint16_t panId = otLinkGetPanId(mOTInst);
-        err            = encoder.Encode(panId);
+        err = encoder.EncodeNullableOnCondition(mIsAttached, [this](const auto pAttrEncoder) -> CHIP_ERROR {
+            uint16_t panId = otLinkGetPanId(mOTInst);
+            return pAttrEncoder->Encode(panId);
+        });
     }
     break;
 
     case ThreadNetworkDiagnostics::Attributes::ExtendedPanId::Id: {
-        const otExtendedPanId * pExtendedPanid = otThreadGetExtendedPanId(mOTInst);
-        err                                    = encoder.Encode(Encoding::BigEndian::Get64(pExtendedPanid->m8));
+        err = encoder.EncodeNullableOnCondition(mIsAttached, [this](const auto pAttrEncoder) -> CHIP_ERROR {
```

Comment:
A much cheaper way would be to do a `if (!mIsAttached) { return EncodeNull(); }` or surch.

Reply:
I'll compare code size, I jsut wanted to avoid a bunch of addition if else in that use case. But sure, if there is a big size impact and can change a if in each case

Reply:
I applied what we discussed on slack